pyccel/parser/semantic.py

```diff
@@ -2502,9 +2503,12 @@ def _visit_FunctionDef(self, expr, **settings):
 
             results_names = [str(i) for i in results]
 
-            assigned = get_assigned_symbols(body)
+            assigned = get_assigned_symbols(body, False)
             assigned = [str(i) for i in assigned]
 
+            all_assigned = get_assigned_symbols(body, True)
+            all_assigned = [str(i) for i in all_assigned]
```

Comment:
If we need the result for `all=True` and `all=False` would it not be simpler to do:
```
all_assigned = get_assigned_symbols(body, True)
assigned      = [a for a in all_assigned if a.rank >0]
```
?
This way we don't need to parse the tree twice

Reply:
Is the `all` argument still necessary with this improvement?

Reply:
no, no need for the `all` argument now.